v3.32.0

bittiez opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced an Assistant menu for managing auto loot, auto sell, auto buy, and mobile graphic filters in a unified interface.
  - Added a simple progress bar UI control, available for Python API gumps.
  - Implemented a quick spell search gump for fast spell access.
  - Added ability to add and remove user markers on the world map.
  - Added script toggling and map marker management to the scripting API.
  - Added HTML crash log generation for improved crash reporting.
  - Added new scripting API methods: CloseGumps, HasTarget, AutoLootContainer, CreateSimpleButton, CreateGumpSimpleProgressBar, ToggleScript, AddMapMarker, RemoveMapMarker.
  - Added a new command to update the scripting API in-game.

- **Improvements**
  - Moved assistant-like options from the options menu to a dedicated Assistant menu.
  - Enhanced auto loot and corpse looting features, including forced container looting and improved duplicate prevention.
  - Increased minimum drag distance for gumps for better usability.
  - Improved scrollbar handling and UI responsiveness in various controls.
  - Added lock/unlock functionality for the counter bar gump, with Alt+Ctrl+Click shortcut and persistent state.
  - Added option to keep Python API-created gumps open after script stop.
  - Improved error handling and profiling scopes for UI updates and input.
  - Refined skill gump interaction with double-click drag initiation.
  - Updated gump parsing to support additional pic-in-pic variants with hue support.
  - Added control method to set position coordinates simultaneously.
  - Updated Discord invite link in README with graphical banner.
  - Always register unhandled exception handler and generate HTML crash logs on crashes.
  - Enhanced equipped item updates by requesting OPL data automatically.
  - Added global access to item move queue instance.
  - Adjusted item container traversal logic for improved filtering.
  - Improved internal profiling data accessibility.
  - Added lock state save/load and event on gump lock changes.

- **Bug Fixes**
  - Fixed issues with the nearby corpse loot gump, including scrollbar and corpse processing.
  - Various minor bug fixes and stability improvements.

- **Documentation**
  - Expanded and clarified scripting API documentation, including new methods and usage examples.
  - Updated README and in-game documentation for new features and commands.
  - Added documentation for new control positioning method.

- **Chores**
  - Incremented version to 3.32.0.
  - Updated internal profiling and error handling for better diagnostics.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

bittiez commented:
You need to use a setter var for it to work, it is functioning as intended.

Make sure the action passed to it is not a straight variable, needs to be a public setter
```
private ushort hue = 0;
public ushort Hue { get { return hue; } set { hue = value; }
```

elderwyn commented:
I see you went with the subscribe method which I prefer more. We want to refactor and remove Actions<> and implement subscribing internally on Gump

I will update my branch to latest.
![image](https://user-images.githubusercontent.com/127065794/228700940-dd19ec36-e459-433b-a835-6c0fbea3ff3d.png)

bittiez commented:
hueChanged is already invoke, no need to invoke in the setter also

bittiez committed:
```diff
+ Added simple progress bar to python API
+ Generate html crash reports and open in browser
```

bittiez committed:
```diff
+ Counter bar is now more easily lockable with ctrl + alt + click
```

bittiez committed:
```diff
+ Automatically grab OPL data from equipped items
```

bittiez committed:
```diff
+ Allow scavenger to work with auto loot disabled
```

bittiez committed:
```diff
+ Python API InJournal and InJournalAny can now use regex if prepended with $
```

bittiez committed:
```diff
+ Minor changes to auto loot backend
+ Auto loot also uses the move item queue now
+ Some fixes for the scroll area of the nearby corpse loot gump
```

bittiez committed:
```diff
+ Crash fix for HtmlControl
+ More nearby corpse gump changes to fix weird scrolling issues
```

bittiez committed:
```diff
+ Added control.SetPos(x, y) for python API gump usage
+ Added CreateSimpleButton for python API gumps
```

bittiez committed:
```diff
+ Added `-updateapi`, this will download the latest DEV version of API.py into your LegionScripts folder for python API scripting.
```

bittiez committed:
```diff
+ Very slightly increased minimum distance to start dragging a gump. Hopefully it should prevent accidental drags instead of clicks.
+ Nameplates should stay below other UI elements now.
+ Nearby loot gump now stays open after relogging.
+ Updated README with discord banner.
```

bittiez committed:
```diff
+ Moved some assistant-like options to their own menu. Access from Top Bar -> More +
```

bittiez committed:
```diff
- Advanced skill gump is easier to drag. Double click skill names for skill button.
```

bittiez committed:
```diff
+ XML Gumps save locked status now. (Ctrl + Alt + Click to lock)
+ Python API created gumps will automatically close when the script stops, unless marked keep open.
```

bittiez committed:
```diff
+ API.py dummy file now has method descriptions/examples
```

bittiez committed:
```diff
+ Added togglescript command to LScript
```

bittiez committed:
```diff
+ Nearby Corpse gump will hue corpse yellow after they have been opened
+ Continue to try to open corpses until they have been opened
```

bittiez committed:
- Added some debug and crash prevention for crash suspected to be caused by RE

bittiez committed:
```diff
+ Added API.HasTarget() to python scripting
+ Added API.ToggleScript() to python scripting
```

bittiez committed:
```diff
+ Added better error handling for bad compressed gump packet.
+ Better error handling playing a legion script that that was unable to generate.
```

bittiez committed:
Started work on spell bar stuff

bittiez committed:
Support for new gump methods

bittiez committed:
```difff
+ Fix packet bug causing gradual fps drop with excessive packets received from server.
+ Better profiling for future debugging.
+ Minor fixes.
```

bittiez committed:
```diff
+ Quick spell cast gump now casts when used with -ss command
+ Rearranged some top menu bar items
+ Quick cast gump will cast when pressing enter after typing
```

bittiez committed:
```diff
- Removed -ss command, open quick spell gump from top menu -> more
+ Added AddMapMarker, RemoveMapMarker, AutoLootContainer to python API
+ Other minor changes
```

bittiez committed:
Prepare for next version on `main` branch

bittiez committed:
```diff
+ Moved Python API CloseGumps to use main thread.
+ Implementing API.Found for various methods that search for items or mobiles.
+ Added hue to HeadMsg
```

bittiez committed:
Fix for recent python API change

bittiez committed:
One more fix

bittiez committed:
Add recursive option to Python API

bittiez committed:
Fix skill changed message not saving in options
